feat(changelog): secret-manager-added-secrets-can-be-set-as-ephemeral 2024-01-04 (#2583)
scaleway/docs-content: changelog/january2024/2024-01-04-secret-manager-added-secrets-can-be-set-as-ephemeral.mdx

---
title: Secrets can be set as ephemeral
status: added
author:
    fullname: 'Join the #secret-manager channel on Slack.'

>     url: 'https://slack.scaleway.com/T7YEXCR7X/C04KGMME3U1'

Reviewer: ```suggestion
    url: 'https://slack.scaleway.com'
```

date: 2024-01-04
category: security-identity
product: secret-manager
---

> Ephemeral secrets are available on Secret Manager API.

Reviewer: ```suggestion
Ephemeral secrets are now available on the Secret Manager API.
```

> At their creation, Secrets can now be set up with a time-to-live or an expiration after a single access. All secrets versions will inherit from this ephemeral policy.

Reviewer: ```suggestion
Upon creation, secrets can now be configured with a time-to-live or to expire after a single access through an ephemeral policy. If applied on a secret, all its versions will inherit from the policy.
```

> Refer to the [Secret Manager API](https://www.scaleway.com/en/developers/api/secret-manager) for more information.

Reviewer: ```suggestion
Refer to the [Secret Manager API documentation](https://www.scaleway.com/en/developers/api/secret-manager) for more information.
```
